ABB.h

```diff
@@ -0,0 +1,538 @@
+    cout << "\n";
+}
+
+////////////////////////////////////////////////
+template <typename Tipo>
+void ABB<Tipo>:: recorridoEnAncho() {
+    recorridoEnAncho(raiz);
+    cout << "\n";
+}
+
+////////////////////////////////////////////////
+template <typename Tipo>
+void ABB<Tipo>:: mostrarData() {
+
+    cout << "\n\n\t__________________________________________\n";
+    cout << "\tIn order: ";
+    recorridoInOrder();
+    cout << "\tPre order: ";
+    recorridoPreOrder();
+    cout << "\tPost order: ";
+    recorridoPostOrder();
+    cout << "\tEn amplitud: ";
+    recorridoEnAncho();
+
+    cout << "\n\tLa raiz del ABB es: " << raiz->obtenerDato() << "\n";
+    cout << "\tLa altura del ABB es: " << obtenerAltura() << "\n";
+    cout << "\tMax: " << obtenerMax(raiz)->obtenerDato() << "\n";
+    cout << "\tMin: " << obtenerMin(raiz)->obtenerDato() << "\n";
+
+    if (estaBalanceado())
+        cout << "\tEl ABB esta balanceado! " << "\n\n";
+    else
+        cout << "\tEl ABB no esta balanceado! " << "\n";
+
+    cout << "\t__________________________________________\n\n";
+}
+
+/* ------------------------------------------------------------------------------ */
+
+
+/* ------------------------------- Metodos Privados ----------------------------- */
+
+////////////////////////////////////////////////
+template <typename Tipo>
+NodoABB<Tipo>* ABB<Tipo>:: insertar(NodoABB<Tipo>* nodo, Tipo dato) {
+    if (nodo == 0)
+        nodo = new NodoABB<Tipo>(dato);
+
+    else if (dato > nodo->obtenerDato())
+        nodo->asignarDerecha(insertar(nodo->obtenerDerecha(), dato));
+
+    else
+        nodo->asignarIzquierda(insertar(nodo->obtenerIzquierda(), dato));
+
+    return nodo;
+}
+
+////////////////////////////////////////////////
+template<typename Tipo>
+int ABB<Tipo>:: obtenerAltura(NodoABB<Tipo>* nodo) {
+    if (!nodo)
+        return 0;
+    return 1 + max(obtenerAltura(nodo->obtenerIzquierda()), obtenerAltura(nodo->obtenerDerecha()));
+}
+
+////////////////////////////////////////////////
+template<typename Tipo>
+bool ABB<Tipo>:: estaBalanceado(NodoABB<Tipo>* nodo) {
+    if (!nodo)
+        return false;
+    return abs(obtenerAltura(nodo->obtenerIzquierda()) - obtenerAltura(nodo->obtenerDerecha())) <= 1;
+}
+
+////////////////////////////////////////////////
+template<typename Tipo>
+NodoABB<Tipo>* ABB<Tipo>:: borradoHoja(NodoABB<Tipo>* nodo) {
+    delete nodo;
+    return 0;
+}
+
+////////////////////////////////////////////////
+template<typename Tipo>
+NodoABB<Tipo>* ABB<Tipo>:: borradoUnHijo(NodoABB<Tipo>* nodo) {
+
+    if (nodo->soloHijoDerecha()) {
+        NodoABB<Tipo>* nodoDer = nodo->obtenerDerecha();
+        nodoDer->asignarPadre(nodo->obtenerParent());
+        NodoABB<Tipo>* aux = nodo;
+        delete aux;
+        return nodoDer;
+    }
+
+    else if (nodo->soloHijoIzquierda()) {
+        NodoABB<Tipo>* nodoIzq = nodo->obtenerIzquierda();
+        nodoIzq->asignarPadre(nodo->obtenerParent());
```

Comment:
El metodo obtenerParent no exsiste, se les debe haber olvidado cambiar esto cuando hicieron el pasaje de ingles a castellano.

```suggestion
        nodoIzq->asignarPadre(nodo->obtenerPadre());
```

Reply:
Arreglado